terraform-aws-redshift-cluster/main.tf

```diff
@@ -3,20 +3,342 @@
 #  Data blocks next to resources that are referencing them
 #  Reduce hard coded inputs where possible. They are used below for simplicity to show structure
 
-/* local {
-  # Local that is a map that is used for something
-  example-local {
-    key = value
+data "aws_availability_zones" "available" {}
+
+locals {
+  name   = "test-${basename(path.cwd)}"
+
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 3)
+
+  s3_prefix = "redshift/${local.name}/"
+
+  tags = {
+    Example    = local.name
+    GithubRepo = "hashi-demo-lab/tfc-dpaas-demo"
   }
+
 }
 
-data "vault_auth_backend" "kubernetes" {
-  namespace = var.namespace
-  path      = "kubernetes"
+################################################################################
+# Complete
+################################################################################
+
+module "redshift" {
+  source = "terraform-aws-modules/redshift/aws"
+
+  cluster_identifier    = local.name
+  allow_version_upgrade = true
+  node_type             = "ra3.xlplus"
+  number_of_nodes       = 3
+
+  database_name   = "mydb"
+  master_username = "mydbuser"
+  # Either provide a good master password
+  #  create_random_password = false
+  #  master_password        = "******" # Do better!
+  # Or make Redshift manage it in secrets manager
+  manage_master_password = true
```

Comment:
we the outputs to map into datazone. particularly cluster details and secrets arn https://github.com/terraform-aws-modules/terraform-aws-redshift/blob/master/outputs.tf.
See screenshot in lucid 3 parameters